pulp_container/app/registry_api.py

```diff
@@ -1008,18 +1008,24 @@ def put(self, request, path, pk=None):
             add_content_units = [str(manifest.pk)]
             remove_content_units = []
 
-        dispatched_task = dispatch(
+        immediate_task = dispatch(
             add_and_remove,
             exclusive_resources=[repository],
             kwargs={
                 "repository_pk": str(repository.pk),
                 "add_content_units": add_content_units,
                 "remove_content_units": remove_content_units,
             },
+            immediate=True,
+            deferred=False,
         )
 
-        if has_task_completed(dispatched_task):
+        if immediate_task.state == "completed":
```

Comment:
historically we've been deleting all tasks involved in docker push, whether it completed or errored https://github.com/pulp/pulp_container/blob/main/pulp_container/app/utils.py#L134
This aspect however, will go away once we implement 'pending blobs', until then we should probably continue deleting those tasks